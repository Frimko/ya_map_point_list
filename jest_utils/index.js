export const createStateTransitionTesterReducers = ({ reducer, initialState }) => {
  return (testCaseName, data = { modifiedState: initialState, action: { type: '' } }) => {
    const { modifiedState, action } = data;
    test(testCaseName, () => {
      if (!action) {
        expect(modifiedState).toMatchSnapshot();
      } else {
        expect(reducer(modifiedState, action)).toMatchSnapshot();
      }
    });
  };
};
