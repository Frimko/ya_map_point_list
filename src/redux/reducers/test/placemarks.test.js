import { createStateTransitionTesterReducers } from 'jest_utils';
import reducer from '../placemarks';

const initialState = [
  {
    id: 'lllsss7',
    coordinates: [
      55.67938453142209,
      37.47520507812455,
    ],
    name: 'Это метка2',
  },
  {
    id: 'ggg88',
    coordinates: [
      55.796702547976345,
      37.782572021483865,
    ],
    name: 'Это метка1',
  },
];

describe('test auth reducer', () => {
  const testStateTransition = createStateTransitionTesterReducers({ reducer, initialState });

  describe('snapshots', () => {
    testStateTransition('initial state');

    testStateTransition('setNewMark', {
      action: {
        type: 'SET_NEW_MARK',
        payload: {
          name: 'новая метка 666',
        },
      },
      modifiedState: [
        ...initialState,
      ],
    });

    testStateTransition('deleteMark', {
      action: {
        type: 'DELETE_MARK',
        payload: {
          id: '5566f',
        },
      },
      modifiedState: [
        ...initialState,
        {
          id: '5566f',
          coordinates: [
            55.796702547976345,
            37.782572021483865,
          ],
          name: 'Это метка1',
        },
      ],
    });
    testStateTransition('changeSortMark', {
      action: {
        type: 'CHANGE_SORT_MARK',
        payload: {
          data: [
            {
              id: 'ggg88',
              coordinates: [
                55.796702547976345,
                37.782572021483865,
              ],
              name: 'Это метка1',
            },
            {
              id: 'lllsss7',
              coordinates: [
                55.67938453142209,
                37.47520507812455,
              ],
              name: 'Это метка2',
            },
          ],
        },
      },
      modifiedState: [
        {
          id: 'lllsss7',
          coordinates: [
            55.67938453142209,
            37.47520507812455,
          ],
          name: 'Это метка2',
        },
        {
          id: 'ggg88',
          coordinates: [
            55.796702547976345,
            37.782572021483865,
          ],
          name: 'Это метка1',
        },
      ],
    });

    testStateTransition('changeCoordinatesMark', {
      action: {
        type: 'CHANGE_SORT_MARK',
        payload: {
          id: 'lllsss7',
          coordinates: [
            666,
            777,
          ],
        },
      },
      modifiedState: [
        {
          id: 'lllsss7',
          coordinates: [
            55.67938453142209,
            37.47520507812455,
          ],
          name: 'Это метка2',
        },
        {
          id: 'ggg88',
          coordinates: [
            55.796702547976345,
            37.782572021483865,
          ],
          name: 'Это метка1',
        },
      ],
    });
  });
});
