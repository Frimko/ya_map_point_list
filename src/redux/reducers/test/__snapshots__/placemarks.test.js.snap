// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test auth reducer snapshots changeCoordinatesMark 1`] = `undefined`;

exports[`test auth reducer snapshots changeSortMark 1`] = `
Array [
  Object {
    "coordinates": Array [
      55.796702547976345,
      37.782572021483865,
    ],
    "id": "ggg88",
    "name": "Это метка1",
  },
  Object {
    "coordinates": Array [
      55.67938453142209,
      37.47520507812455,
    ],
    "id": "lllsss7",
    "name": "Это метка2",
  },
]
`;

exports[`test auth reducer snapshots deleteMark 1`] = `
Array [
  Object {
    "coordinates": Array [
      55.67938453142209,
      37.47520507812455,
    ],
    "id": "lllsss7",
    "name": "Это метка2",
  },
  Object {
    "coordinates": Array [
      55.796702547976345,
      37.782572021483865,
    ],
    "id": "ggg88",
    "name": "Это метка1",
  },
]
`;

exports[`test auth reducer snapshots initial state 1`] = `
Array [
  Object {
    "coordinates": Array [
      55.67938453142209,
      37.47520507812455,
    ],
    "id": "lllsss7",
    "name": "Это метка2",
  },
  Object {
    "coordinates": Array [
      55.796702547976345,
      37.782572021483865,
    ],
    "id": "ggg88",
    "name": "Это метка1",
  },
]
`;

exports[`test auth reducer snapshots setNewMark 1`] = `
Array [
  Object {
    "coordinates": Array [
      55.67938453142209,
      37.47520507812455,
    ],
    "id": "lllsss7",
    "name": "Это метка2",
  },
  Object {
    "coordinates": Array [
      55.796702547976345,
      37.782572021483865,
    ],
    "id": "ggg88",
    "name": "Это метка1",
  },
  Object {
    "coordinates": Array [
      55.76,
      37.64,
    ],
    "id": 777,
    "name": "новая метка 666",
  },
]
`;
